src/derivkit/forecasting/forecast_core.py

```diff
@@ -0,0 +1,237 @@
+            differentiate. It should accept a list or array of parameter
+            values as input and return either a scalar or a
+            :class:`np.ndarray` of observable values.
+        theta0: The points at which the
+            derivative is evaluated. A 1D array or list of parameter values
+            matching the expected input of the function.
+        cov: The covariance matrix of the observables. Should be a square
+            matrix with shape (n_observables, n_observables), where n_observables
+            is the number of observables returned by the function.
+        forecast_order: The requested order D of the forecast:
+
+                - D = 1 returns a Fisher matrix.
+                - D = 2 returns the 3-d and 4-d tensors required for the
+                  doublet-DALI approximation.
+                - D = 3 would be the triplet-DALI approximation.
+
+            Currently only D = 1, 2 are supported.
+        method: Method name or alias (e.g., "adaptive", "finite"). If None,
+            the DerivativeKit default ("adaptive") is used.
+        n_workers: Number of workers for per-parameter parallelization/threads.
+            Default 1 (serial). Inner batch evaluation is kept serial to avoid
+            nested pools.
+        **dk_kwargs: Additional keyword arguments passed to DerivativeKit.differentiate.
+
+    Returns:
+        If ``D = 1``: Fisher matrix of shape ``(P, P)``.
+        If ``D = 2``: tuple ``(G, H)`` with shapes ``(P, P, P)`` and ``(P, P, P, P)``.
+
+    Raises:
+        ValueError: If `forecast_order` is not 1 or 2.
+
+    Warns:
+        RuntimeWarning: If `cov` is not symmetric (proceeds as-is, no symmetrization),
+            is ill-conditioned (large condition number), or inversion
+            falls back to the pseudoinverse.
+    """
+    if forecast_order not in (1, 2):
+        raise ValueError(
+            "Only Fisher (order 1) and doublet-DALI (order 2) forecasts are currently supported."
+        )
+
+    theta0_arr = np.atleast_1d(theta0)
+    cov_arr = validate_covariance_matrix(cov)
+    n_observables = cov_arr.shape[0]
+
+    # Check model output dimension
+    y0 = np.atleast_1d(function(theta0_arr))
+    if y0.shape[0] != n_observables:
+        raise ValueError(
+            f"Expected {n_observables} observables from model (from cov {cov.shape}), "
+            f"but got {y0.shape[0]} (output shape {y0.shape})."
+        )
+
+    # Compute inverse covariance matrix
+    invcov = invert_covariance(cov_arr, warn_prefix="get_forecast_tensors")
+
+    # Compute first-order derivatives
+    d1 = _get_derivatives(
+        function,
+        theta0_arr,
+        cov_arr,
+        order=1,
+        n_workers=n_workers,
+        method=method,
+        **dk_kwargs,
+    )
+
+    if forecast_order == 1:
+        return np.einsum("ai,ij,bj->ab", d1, invcov, d1)  # Fisher
+
+    # Compute second-order derivatives
+    d2 = _get_derivatives(
```

Comment:
```suggestion
    second_order_derivative = _get_derivatives(
```
In any case the comment seems redundant.